npm/deploy/Options.kt

```diff
@@ -0,0 +1,73 @@
+package com.vaticle.bazel.distribution.npm.deploy
+
+import com.vaticle.bazel.distribution.npm.deploy.Options.CommandLineParams.NPM_PATH
+import com.vaticle.bazel.distribution.npm.deploy.Options.CommandLineParams.RELEASE_REPO
+import com.vaticle.bazel.distribution.npm.deploy.Options.CommandLineParams.SNAPSHOT_REPO
+import com.vaticle.bazel.distribution.npm.deploy.Options.Env.DEPLOY_NPM_TOKEN
+import com.vaticle.bazel.distribution.npm.deploy.Options.RepositoryType.RELEASE
+import com.vaticle.bazel.distribution.npm.deploy.Options.RepositoryType.SNAPSHOT
+import picocli.CommandLine
+
+class Options {
+    @CommandLine.Option(names = [NPM_PATH], required = true)
+    lateinit var npmPath: String
+
+    @CommandLine.Option(names = [SNAPSHOT_REPO], required = true)
+    private lateinit var snapshotRepo: String
+
+    @CommandLine.Option(names = [RELEASE_REPO], required = true)
+    private lateinit var releaseRepo: String
+
+    @CommandLine.Parameters
+    private lateinit var params: List<String>
+
+    val registryURL: String
+        get() {
+            if (params.isEmpty() || params[0].isBlank()) {
+                throw IllegalArgumentException("Missing required positional argument: <${RepositoryType.allValuesString}>")
+            }
+            return when (RepositoryType.of(params[0])) {
+                SNAPSHOT -> snapshotRepo
+                RELEASE -> releaseRepo
+            }
+        }
+
+    val npmToken: String
+        get() {
+            return System.getenv(DEPLOY_NPM_TOKEN)
+                ?: throw IllegalArgumentException("token should be passed via \$${DEPLOY_NPM_TOKEN} env variable")
+        }
+
+    companion object {
+        fun of(args: Array<String>): Options {
+            val commandLine = CommandLine(Options())
+            val parseResult: CommandLine.ParseResult = commandLine.parseArgs(*args)
+            assert(parseResult.asCommandLineList().size == 1)
```

Comment:
why can we guarantee that there's exactly one element returned by `asCommandLineList`?

Also since there's no guarantee that `asCommandLineList()` is hermetic (it most likely is, but we can't say for sure), this is the better style:
```
val cliList = parseResult.asCommandLineList()
assert (cliList.size == 1)
return cliList[0].getCommand()
```

Reply:
See the documentation for `asCommandLineList`

> Returns this ParseResult as a list of CommandLine objects, one for each matched command/subcommand.
>    * Note that for repeatable subcommands, there may be multiple commands at each level of the hierarchy in the returned list.